### DISCUSSION TITLE
2 MPPTs bei Ausgangs-Limit berücksichtigen

### DISCUSSION THREAD

**Discussion by ulkrueger** [tbnobody/OpenDTU#811] [Ideas]
Leider wird bei einem Hoymiles WR mit 2 MPPTs bei einem eingestellten Limit von 600W an jedem MPPT nur max 300W zugelassen. Hier wäre eine dynamische Regelung wünschenswert die auch 600Watt Ausgangsleistung ermöglichen, wenn MPPT1 350W und MPPT2 250W bringen. Ohne zusätzliche Regelung würden jetzt nur 550W Ausgangsleistung erbracht.

Die gewünschte Regelung müßte also immer zunächst das Limit auf 800W setzen und falls die tatsächliche Ausgangsleistung 600W übersteigt, die Limitierung herabsetzen. Umgekehrt müßte bei einem Ausgangswert unter 600W die Limitierung wieder auf > 600 gesetzt werden.

**Comment by homeautomation2022** [comment]
Technisch möglich ja, ich sag mal in min. 15s Regelung(besser noch weniger), damit die Funkverbindung nicht komplett zeitlich gestört wird. Das macht einen HM-800 aber auch nicht zu einem in DE legalen HM-600, da es immernoch Spitzen von bis zu 800W gibt.

**Comment by NF1971** [comment]
600 Watt Limit heißt beim HM-1500 2x300 Watt Limit (300 rechte Anschlüsse / 300 linke Anschlüsse). Ich habe vier Module (2 Ost / 2 West) und obwohl ich erst Bedenken hatte,  je 1x Ost/West auf der rechten und linken Seite angeschlossen; das kann er :-) Dann klappt es auch mit dem 600er Limit.

**Reply by ulkrueger** [reply]
Interessante Idee. Ich fürchte nur eine gute MPPT-Regelung kann ich dann vom Wechselrichter nicht mehr erwarten, wenn z.B. an einem MPPT-String gemeinsam ein beschattetes und eine beschienenes Modul hängen? Funktionieren wird das sicher, aber läuft das dann auch effizient?

**Comment by NF1971** [comment]
Wie gesagt, Bedenken hatte ich auch. Ich habe vorgestern meine beiden Ost-Module aufgebaut und beide so angeschlossen, wie ich es für richtig gehalten habe (beide rechts).
Ergebnis: Kurz vor 9.00 wurden 300 Watt erreicht und er hat ab geregelt, obwohl das Limit auf 600 Watt eingestellt war.

![PV1](https://user-images.githubusercontent.com/116456499/232771376-c7f66da9-4a9b-424d-8ff9-abcf3f0f11f5.jpg)

Gestern Nachmittag haben wir dann zusätzlich die Westmodule installiert und die Verkabelung geändert (1xOst/West links, 1xOst/West rechts). Heute Morgen sah es besser aus:

![PV2](https://user-images.githubusercontent.com/116456499/232771498-d2f158b9-1e34-4a23-9dfc-6f697d99bf24.png)

Die Sonne wandert und die Werte verschieben sich von Ost nach West. Optimal…..

![PV3](https://user-images.githubusercontent.com/116456499/232771510-78cc6aab-c7e5-4842-93da-b8ad30ccd88c.jpg)
Der unterschiedliche Gesamtertrag ist übrigens entstanden, weil ich erst nur meine zwei alten Module auf Ost/West liegen hatte, dann einen Tag beide auf Ost.
Ich habe leider keinen Hinweis gefunden, wie man die Zähler beim HM zurücksetzen kann.

**Reply by Solarteur** [reply]
> Ich habe leider keinen Hinweis gefunden, wie man die Zähler beim HM zurücksetzen kann.

Die Zähler in den HM-Wechselrichtern können (über OpenDTU) nicht zurückgesetzt werden. Über "Wechselrichter Einstellungen" --> "Wechselrichterliste" und dort "Wechselrichter bearbeiten" kann auf String-Ebene ein "Ertragsversatz" eingegeben werden.

**Reply by NF1971** [reply]
Danke für den Tipp. 
Einmal die Werte von 16:20. Der Ertrag kommt jetzt überwiegend von den West-Modulen. Aus meiner Sicht bekommt der Wechselrichter das gut hin. 
 
![PV4](https://user-images.githubusercontent.com/116456499/232807169-ea2b705f-9e53-4a41-bd60-051069c84e51.jpg)

**Reply by ulkrueger** [reply]
Das wäre ja eine sehr elegante Methode um auf einfachste Art die 600W-Limitierung auszureizen. Werd ich auch mal ausprobieren. Vielen Dank für den Trick und die Messwerte!

**Reply by baba-joe** [reply]
Moin, laut EEG beziehen sich Leistungsangaben einer Solaranlage immer auf die verbauten Module (Holger Laudeley, https://www.youtube.com/watch?v=vstYMn4cTIA ab ca. 8min 25s). Die 600W waren wohl ein Vorschlag, um weitgehend "genehmigungsfrei" über Schuko oder ähnliche Steckverbinder direkt ins Hausnetz einzuspeisen. Jetzt gibt es einen neuen Vorschlag des VDE (https://www.vde.com/de/presse/pressemitteilungen/2023-01-11-mini-pv), eine Bagatellgrenze von 800W auf das steckerfertige Gesammtsystem einzuführen. 
Abseits irgend welcher möglicher gesetzlichen Auflagen habe ich an meinen Wechselrichtern HM-1500 (dank openDTU) mal getestet, wie das Regelverhalten aussieht. Anscheinend gibt es die zwei MPPT-Regelkreise für jeweils String 1+2 und 3+4, was bedeutet, das man pro Regelkreis komplett unterschiedliche Module paarweise verbauen kann, so lange sie innerhalb der Spezifikation des WR liegen. Dazu kommt aber noch einen Art "Schattenmanagement", was eingreift und jedes Modul separat runterregelt, ohne das der "Stringpartner" auch runtergezogen wird. Bei meinem Test habe ich vier identische Module gleich ausgerichtet und jeweils eins Teilverschattet, ohne das sich die anderen in der Leistung verändert haben. Das ist schon Klasse und vermutlich funktioniert deshalb das Setup mit gemischter Ausrichtung. Wie schnell eine gesetzte Limiterung tatsächlich vom WR umgesetzt wird kann man allerdings mit der openDTU nicht präzise bestimmen, da hilft eine externe Leistungsmessung, die eine wesentlich höhere Messauflösung bietet. Die Hoymiles liefern bei mir trotz guter Verbindung anscheinend nur selten Daten, oft weniger als ein Messwert pro Minute?

**Reply by helgeerbe** [reply]
> Die Hoymiles liefern bei mir trotz guter Verbindung anscheinend nur selten Daten, oft weniger als ein Messwert pro Minute?

Wie viele Hoymiles hast du denn angeschlossen? Ich habe 2 Hoymiles und eine Poll-Rate von 5s eingestellt. Auf der Oberfläche von OpenDTU habe ich maximal nach 12 Sekunden eine Aktualisierung der Daten.

**Reply by baba-joe** [reply]
Hab aktuell 6 angeschlossen, die DTU steht draussen in max. 5m Entfernung Luftlinie. Hab die Version mit Antennen und die Leistung auf hoch. Pollrate habe ich von 1s bis 30s ausprobiert, macht aber wenig Unterschied bei mir. Gerade ist die auf 5s gesetzt, die WR liefern aber etwa alle 30s Daten. Gestern hatte ich noch 7 WR eingebunden, da waren es auch öfter mal über 100s. Der Entwickler tbnobody meinte sinngemäss, dass nur Daten kommen, wenn der WR auch liefert. Das scheint mir allerdings irgendwie unlogisch, wozu gibt es dann die einstellbare Pollrate? Ob die Daten häufiger kommen, wenn man weniger WR eingebunden hat werde ich mal testen. Die 5s sind ja schon nicht so prickelnd, wenn ich das mit meinen anderen Geräten vergleiche, die übers Victron Cerbo GX eingebunden sind. Hab gestern mal das henne49 script auf dem Cerbo ausgeführt und da kann man schön sehen, wie langsam die Hoymiles Daten ändern. Zum Vergleich habe ich noch eine Leistungsmessung mitlaufen, die quasi sofort reagiert und da sind zum Teil erhebliche Unterschiede, wenn das Wetter auf Sonne-Wolken Mix steht.

**Reply by baba-joe** [reply]
Die Anzahl der eingebundenen WR spielt definitv eine Rolle. Die Formel scheint folgende zu sein: Anzahl der eingebunden WR mal eingestellter Pollingzeit = die echte Zeit bis Daten aktualisiert werden. Habe mit 6 WR und 5s angefangen und meistens kam eine Akualisierung nach ca. 30s. Danach 3 WR und 5s mit ca. 15s, 2 WR und 5s mit 10s, 2 WR und 1s mit 2s und 1 WR auf 1s alle s eine Aktualisierung. Bei 6 WR und 4s Polling sind es ca. 25s für eine Aktualisierung, gehe ich runter auf 3s Polling bei 6 WR scheint ein Limit überschritten und die Aktualisierungszeit geht extrem hoch. Das klingt mir eher nach einem Bug als nach einem WR-Problem?
![data_age2](https://user-images.githubusercontent.com/130974809/233390986-05339b6d-8ab0-4bb9-9e39-b19b58f8fdbc.png)

**Reply by tbnobody** [reply]
- Lesen Inverter 1
- Warten Poll Interval
- Lesen Inverter 2
- Warten Poll Interval
- Lesen Inverter 3
- Warten Poll Interval
- Lesen Inverter 1
- Warten Poll Interval
- etc....

und die Zeit zwischen dem Lesen wird zum abarbeiten von Mqtt etc. verwendet

**Reply by baba-joe** [reply]
Ok, das erklärt einiges, Danke. Warum geht dann aber das System in eine Art "Dauerpoll", wenn man bestimmte Konfigurationen nutzt? Z. B. diese 6 WR mit 3s Polling führen zu diesen endlos langen Polls. Ausserdem hatte ich den Eindruck, dass das Web UI teilweise nicht mehr reagiert, wenn man z. B. das Pollling auf 1s bei 6 WR stellt? Und läuft man nicht in ein Problem, wenn man bei 10 WR das Polling auf 5s stehen hat, mal abgesehn davon, dass 50s auch schon ziemlich lange ist. Liegt die Art der Abfrage am Aufbau des Codes oder ist das eine Limitierung der Leistung der Hardware?

**Comment by tknerr** [comment]
Hmmm... ich sehe das bei meinem HM-1500 auch. Mit Drosselung auf 600W wird einfach jeder MTTP Ausgang statisch auf max 300W gedrosselt. Bei einem Ost/West Setup ist das natürlich super ungünstig, wenn die Ost Module nur noch 100W liefern, die West Module aber noch 500W liefern würden, wenn sie nicht auf hart 300W gedrosselt wären... :-/

<img width="1512" alt="image" src="https://github.com/tbnobody/OpenDTU/assets/365744/1ada6e98-b2e5-4b7a-bd64-25b46b200e19">

(hier zu erkennen, zwischen 09:00-11:00 und 15:00-18:00 wären wohl locker 600W Gesamtleistung drin gewesen, wäre nicht jeweils die eine Modulseite / MPPT Ausgang stupide hart auf das 300W Limit gedrosselt gewesen. Das ist echt mega ärgerlich.)

Ist das ein "Feature" vom HM-1500 WR, oder eher ein Bug?

Imho müsste das doch besser Firmware-seitig auf WR Ebene behoben werden, anstatt zu versuchen eine Lösung aussenrum zu basteln... hat das mal jemand an Hoymiles gemeldet? Besteht da Hoffnung dass das von Hoymiles verbessert wird?

**Comment by Marc9485** [comment]
Warum macht ihr die Verteilung nicht einfach anders; String 1 & 3 auf Ost und 2 & 4 auf West. Dann ist die erzeugte Energie auf beide Tracker gleichmäßig verteilt.

**Reply by avandorp** [reply]
Aber hat der HM-1500 nicht nur 2 MPTTs? Das gibt sicher auch Verluste, wenn da unterschiedlich ausgerichtete Panels anliegen. Für mich ist das ein Fehler in der Firmware. Das Limit müsste für den ganzen WR gesetzt werden können.

**Reply by Marc9485** [reply]
Der größte Verlust liegt darin sich nen 1500 Watt Umrichter zu kaufen und den dann aber auf 600 zu begrenzen

**Comment by tknerr** [comment]
Weiss jemand, ob sich der HMS-1600 (mit 4 MPPT) genauso verhält, und bei einem 600W Limit jeden einzelnen MPPT Ausgang dann stupide auf 150W runterregelt?

**Reply by Thjaffi** [reply]
Moin,

ja tut es.
Der Thread ist zwar schon älter, aber ich habe seit 3 Tagen mein HMS-1600 mit 4 Modulen vorübergehend auf dem Rasen liegen während ich die Montage auf dem Dach vorbereite.

Dabei ist mir aufgefallen, dass die Module alle ein einzelnes Limit bekommen (ca. 214W bei 800W Gesamtleistungsgrenze).
Das ist insofern ärgerlich, da ich zum Grundlast decken + Abends ein bisschen den PC etc. abfangen, nun eigl nur 200W pro Modul erzeugen kann. Wollte 1x Ostsüdost, 1x Süd, 2x West. Werd ich auch weiterhin so montieren, bin aber schon ein bisschen von der Regelung enttäuscht.

Auf dem Rasen liegend trifft der Schatten nun zuerst ein Modul, das daraufhin natürlich nicht mehr volle Leistung bringt, die anderen können das aber nicht ausgleichen, da sie einfach nicht mehr als 214W abgeben dürfen, obwohl sie es könnten (hab ich ausprobiert).

Gibt es irgendeine bekannte Möglichkeit, zwar ein Gesamtlimit von 800W einzustellen, aber die Module eben nicht fix auf 200W zu begrenzen? (direkt mit dem WR wenn möglich)

**Reply by ulkrueger** [reply]
Das musste ich vor einigen Tagen auch wieder bei einem Bekannten erleben - wirklich traurig!
Ich hatte mir mal mit OpenDTU und OpenHAB eine Regelung gebaut, die immer den Netz-Output des Hoymiles kontrolliert.
Steigt die Leistung über 800W dann wurde über MQTT der OpenDTU mitgeteilt, dass sie doch bitte die Leistungsbegrenzung auf 50% setzen soll. Sinkt die Leistung unter 800W wurde in wenigen Sekunden die Leistungsbegrenzung wieder auf 100% gesetzt.
Ich suche auch nach jemandem, der diese Regelung entweder direkt in die OpenDTU integriert, oder das ganze in einem weiteren ESP32 unterbringt, der die Ausgangsleistung kennt (MQTT) und immer dementsprechend die Begrenzung anpaßt.
Auch wenn das sicher nicht 100 prozentig der gesetzlichen Regelung entspricht - es wäre eine _vernünftige_ Umsetzung der 800W-Grenze. Letztendlich wäre es aber viel besser, wenn Hoymiles die Wirkungsweise der Begrenzung flexibler programmieren würde.

**Reply by wanderfalk** [reply]
So eine Regelung muss INNERHALB des Wechselrichters passieren, da die Regelung kontinuierlich erfilgen muss und nicht nur alle x Sekunden. 
In der neuesten Deye-Generation ist das so gemacht. Dafür braucht der Deye ...G4 aber eine höhere Startspannung als die Hoymiles und verschenkt morgens mehr Strom.

Mit den Hoymiles könntest du es so probieren: Dafür brauchst du aber zwei getrennte Wechselrichter:
An einem hängen die Ost-Module, am anderen die West-Module.
Morgens Wechselrichter Ost auf 700 Watt und WR West auf 100 W begrenzen. 
Am späten Vormitrag beide auf 400W (erst 700 auf 400 und dann 100 auf 400 - aber eigentlich sind die MPPT zu träge, als dass der Ertrag anders über 800W ginge) und im Laufe des Nachmittags West auf 700 W und Ost auf 100 W.
Je nach effektiver Modulleistung auch andere Limits.
Das geht aber nur mit 2 getrennten WR und nicht mit einem großen Wechselrichter.

**Reply by ulkrueger** [reply]
Natürlich wäre es schön, wenn die Regelung innerhalb des Wechselrichters erfolgen würde, aber warum sollte eine externe Nachregelung alle 5-10 Sekunden nicht auch möglich sein (das schafft die openDTU). Viele haben ja jetzt leider schon einen Hoymiles-Wechselrichter mit 4 PV-Modulen und suchen nach einer Lösung, um möglichst oft und lange die 600W oder 800W als Ausgangsleistung zu erhalten. Diese Regelung in OpenDTU zu integrieren ist eigentlich kein großer Programmieraufwand (s. meinen  Start-Post).

**Comment by thetruemist** [comment]
Hallo zusammen, bin auf diese Diskussion gestoßen, da ich genau das beschriebene Phänomen an einem HMS-1600-4T beobachte. Ich teste gerade mit einem Panel, später werden es vier. Wenn ich den Wechselrichter auf die legalen 800 W begrenze, begrenzt er in Wirklichkeit jeden einzelnen Eingang auf max 200 W. Und ich wunderte mich, warum mein 450 Wp Panel nicht über 200 W kam, trotz optimaler manueller Ausrichtung... 
Das ist natürlich keine gute Lösung für die Limitierung. Der Wechselrichter sollte schlau genug sein, die *Gesamt*-Leistung zu limitieren, nicht jeden Eingang... Ich habe kein Problem damit, max 800 W einzuspeisen, aber ich habe ein Problem damit, meine Panels auf je 200 W zu limitieren. Das ist Unsinn.

Gibt es da inzwischen eine Lösung, eine Idee?

Dank im Voraus,
 Michael

**Reply by thetruemist** [reply]
Ok, ich fand gerade https://www.photovoltaikforum.com/thread/236667-power-distribution-logik-for-hms-1600-1800-2000-4t/
Ich brauche also eine neue Firmware. Na denn...

**Comment by De-Elsaesser** [comment]
Hallo, ja es gibt ein Update 02.00.04 für die HMS-Reihe. Wer OpenDTU hat, muss zuerst ein original DTU-S wieder anschließen, dann beim Homyles-Support nach dem Update fragen. Die machen das dann innerhalb einiger Tage, dann kann man zurück auf openDTU. Die Programmierung habe ich noch nicht vorgenommen. Claude.ai hat mir einen Vorschlag für eine entsprechende Automation gemacht, muss ich näher anschauen und testen.
